src/visu.f90

```diff
@@ -601,17 +607,25 @@ end function gen_filename
 
   function gen_h5path(filename, num)
 
-    character(len=*), intent(in) :: filename, num
+    character(len=*), intent(in) :: filename
+    integer, intent(in) :: num
 #ifndef ADIOS2
     character(len=*), parameter :: path_to_h5file = "./"
     character(len=(len(path_to_h5file) + len(filename))) :: gen_h5path
     write(gen_h5path, "(A)") path_to_h5file//filename
 #else
     character(len=*), parameter :: path_to_h5file = "../data.hdf5:/Step"
-    character(len=(len(path_to_h5file) + len(num) + 1+ len(filename))) :: gen_h5path
-    write(gen_h5path, "(A)") path_to_h5file//num//"/"//filename
+    character(len=:), allocatable :: gen_h5path
+    gen_h5path = path_to_h5file//int_to_str(num)//"/"//filename
```

Comment:
This statement allocates the character array "gen_h5path" ?

Reply:
See https://github.com/xcompact3d/Incompact3d/pull/94#discussion_r795452179